packages/ckeditor5-image/tests/imagestyle/imagestylecommand.js

```diff
@@ -6,116 +6,404 @@
 import ModelTestEditor from '@ckeditor/ckeditor5-core/tests/_utils/modeltesteditor';
 import ImageStyleCommand from '../../src/imagestyle/imagestylecommand';
 import { setData, getData } from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+import ImageTypeCommand from '../../src/image/imagetypecommand';
 
 describe( 'ImageStyleCommand', () => {
-	const defaultStyle = { name: 'defaultStyle', title: 'foo bar', icon: 'icon-1', isDefault: true };
-	const otherStyle = { name: 'otherStyle', title: 'baz', icon: 'icon-2', className: 'other-class-name' };
+	const onlyBlock = {
+		name: 'onlyBlock',
+		title: 'foo',
+		icon: 'icon-1',
+		modelElement: 'image',
+		class: 'image-style-class'
+	};
+	const onlyInline = {
+		name: 'onlyInline',
+		title: 'foo',
+		icon: 'icon-2',
+		modelElement: 'imageInline',
+		class: 'image-style-class'
+	};
+	const anyImage = {
+		name: 'anyImage',
+		title: 'foo',
+		icon: 'icon-3',
+		class: 'image-style-class'
+	};
+	const defaultInline = {
+		name: 'defaultInline',
+		title: 'foo',
+		icon: 'icon-4',
+		modelElement: 'imageInline',
+		isDefault: true
+	};
+	const defaultBlock = {
+		name: 'defaultBlock',
+		title: 'foo',
+		icon: 'icon-5',
+		modelElement: 'image',
+		isDefault: true
+	};
+
+	let model, schema, command, editor;
+
+	beforeEach( async () => {
+		editor = await ModelTestEditor.create();
+
+		model = editor.model;
+		schema = model.schema;
+		command = new ImageStyleCommand( editor, [
+			anyImage,
+			defaultInline, onlyInline,
+			defaultBlock, onlyBlock
+		] );
+
+		schema.register( 'p', { inheritAllFrom: '$block' } );
+		schema.register( 'media', { allowWhere: '$block' } );
+
+		schema.register( 'image', {
```

Comment:
Let's use the real `Image*Editing` plugin here to test real-life use-cases and stay future-proof.

Reply:
Done.